source/Engine.cpp

```diff
@@ -1315,19 +1316,17 @@ void Engine::ThreadEntryPoint()
 	{
 		{
 			unique_lock<mutex> lock(swapMutex);
-			while(calcTickTock == drawTickTock && !terminate)
-				condition.wait(lock);
+			condition.wait(lock, [this] { return !hasFinishedCalculating || terminate; });
 
 			if(terminate)
 				break;
 		}
 
 		// Do all the calculations.
 		CalculateStep();
-
```

Comment:
Keep this newline. The braces are a block scope separate from the above line and should be easily identified as such.